craft_parts/executor/part_handler.py

```diff
@@ -673,44 +724,41 @@ def _migrate_overlay_files_to_stage(self) -> None:
         list of visible overlay entries, converting overlayfs whiteout files
         and opaque dirs to OCI.
         """
-        stage_overlay_state_path = states.get_overlay_migration_state_path(
-            self._part.overlay_dir, Step.STAGE
-        )
-
-        # Overlay data is migrated to stage only when the first part declaring overlay
-        # parameters is migrated.
-        if stage_overlay_state_path.exists():
-            logger.debug(
-                "stage overlay migration state exists, not migrating overlay data"
-            )
-            return
-
         parts_with_overlay = get_parts_with_overlay(part_list=self._part_list)
         if self._part not in parts_with_overlay:
             return
 
         logger.debug("staging overlay files")
-        migrated_files: set[str] = set()
-        migrated_dirs: set[str] = set()
-
-        # process layers from top to bottom (reversed)
-        for part in reversed(parts_with_overlay):
-            logger.debug("migrate part %r layer to stage", part.name)
-            visible_files, visible_dirs = overlays.visible_in_layer(
-                part.part_layer_dir, part.stage_dir
-            )
-            layer_files, layer_dirs = migration.migrate_files(
-                files=visible_files,
-                dirs=visible_dirs,
-                srcdir=part.part_layer_dir,
-                destdir=part.stage_dir,
-                oci_translation=True,
+
+        # process parts in each partition
+        for partition in self._part_info.partitions or (None,):
```

Comment:
@mr-cal This pattern can be used to avoid different codepaths for partitioned or non-partitioned processing. We can create a `get_partitions()` method that includes the `(None,)` tuple.

Reply:
In the next year or so, I'd like to migrate partitions from an opt-in feature to standard functionality in craft-parts.

This is a good step in that direction, since the same code will be running.

Reply:
I initally tried to use "default" instead of None in the no-partitions map, but there are other parts of the code expecting it to be None. If we're able to use "default" in the no-partitions case we'll be closer to uniform processing.